### PR TITLE
refactor: ReconciliationMaxInterval -> MaxReconciliationInterval

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

**Comment by csviri** [comment]
Should not we just rename this? and put it into migration guide also. It will look quite ugly in the main APIs

**Comment by csviri** [comment]
fair enough :)

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/AnnotationControllerConfiguration.java [hunk 1]
```diff
@@ -39,6 +49,9 @@
 public class AnnotationControllerConfiguration<P extends HasMetadata>
     implements io.javaoperatorsdk.operator.api.config.ControllerConfiguration<P> {
 
+  private static final Logger log =
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/AnnotationControllerConfiguration.java:52]: is this used somewhere?

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Constants.java [hunk 2]
```diff
@@ -16,7 +16,14 @@ public final class Constants {
   public static final String NO_VALUE_SET = "";
   public static final String WATCH_CURRENT_NAMESPACE = "JOSDK_WATCH_CURRENT";
   public static final String WATCH_ALL_NAMESPACES = "JOSDK_ALL_NAMESPACES";
+
+  /**
+   * @deprecated use {@link #NO_MAX_RECONCILIATION_INTERVAL} instead
+   */
+  @Deprecated(forRemoval = true)
   public static final long NO_RECONCILIATION_MAX_INTERVAL = -1L;
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/Constants.java:24]: This can be removed I guess

**Review by csviri — APPROVED** [review]
Just left two things to check

**Comment by metacosm** [comment]
Done.

**Review by csviri — APPROVED** [review]
LGTM

**Pull request merged**